_plugins/events.rb

```diff
@@ -0,0 +1,124 @@
+        dstart = date_or_datetime(event['start'])
+        dend   = date_or_datetime(event['end'])
+
+        # Calendar assumes midnight is at the *start* of the day, so a date-only
+        # range like 01-30 -> 01-31 would only render on the 30th.
+        # So add a day for date-only events
+        if (event['start'].class == Date) && !dend.nil?
+          dend   = dend.next_day
+        end
+
+        # Calendar requires JSON like '2016-04-06T15:45:00+01:00' or '2016-04-06'
+        event['start'] = dstart.to_s
+        event['end']   = dend.to_s unless dend.nil?
+        event
+      end
+    end
+
+    def format_event_date start_date, end_date
+      # Compares start_date with end_date to decide how to format the date correctly
+      return "" unless start_date
+
+      if start_date.class == Date
+        # no times, just use dates
+        if end_date.nil? || start_date == end_date
+          # one-day event
+          return start_date.strftime('%-d %B, %Y')
+        else
+          return "#{start_date.strftime('%-d %B, %Y')} - #{end_date.strftime('%-d %B, %Y')}"
+        end
+      else
+        # times included, passed in as strings. more parsing to do
+        dstart = date_or_datetime(start_date)
+        dend   = date_or_datetime(end_date)
+
+        if dend.nil? || dstart.to_date == dend.to_date
+          # one-day event
+          if (dend.nil? || dstart.strftime('%H:%M') == dend.strftime('%H:%M'))
+            # single time
+            return dstart.strftime('%-d %B, %Y - %H:%M (%Z)')
+          else
+            return "#{dstart.strftime('%-d %B, %Y')} - #{dstart.strftime('%H:%M')}-#{dend.strftime('%H:%M (%Z)')}"
+          end
+        else
+          # multiday, times not so relevant (or no endtime set)
+          return "#{dstart.strftime('%-d %B, %Y')} - #{dend.strftime('%-d %B, %Y')}"
+        end
+      end
+
+    end
+
+    def generate_ics(events)
+      cal = Icalendar::Calendar.new
+      cal.x_wr_calname = 'Foreman Community Calendar'
+
+      events.each do |event|
+        cal_event = Icalendar::Event.new
+
+        speaker_text     = event['speaker'].nil? ? '' : "Speaker: #{event['speaker']}\n\n"
+        description_text = event['description'].strip
+
+        cal_event.summary     = event['title'].gsub(/:/, '\\:')
+        cal_event.location    = event['location']
+        cal_event.description = (speaker_text + description_text).strip
+
+        if event['start'] && event['end']
+          cal_event.dtstart = ical_time(event['start'])
+          cal_event.dtend   = ical_time(event['end'])
+        elsif event['start'] && event['start'].class != Date
+          # Has a time, assume 1hr duration
+          cal_event.dtstart = ical_time(event['start'])
+          cal_event.dtend   = ical_time(event['start'].to_datetime + 1.hours)
+        end
+        cal.add_event(cal_event)
+      end
+      cal.to_ical
+    end
+
+  end
+
+end
+
+Liquid::Template.register_filter(Jekyll::Events)
```

Comment:
I guess that raises two questions:
(1) We don't deploy on GitHub pages, so why do we care about that? The rendering clearly works
(2) If we do care, how else can this extra code be loaded? It's pretty substantial, I wouln't want to try and do it in Liquid tags.